utils.go

```diff
@@ -125,25 +124,18 @@ func stringsToAddrInfos(peers []string) ([]peer.AddrInfo, error) {
 	return peer.AddrInfosFromP2pAddrs(madders...)
 }
 
-// generateKey generates an ed25519 private key and writes it to the data directory
-// If the seed is zero, we use real cryptographic randomness. Otherwise, we use a
-// deterministic randomness source to make keys the same across multiple runs.
-func generateKey(seed int64, fp string) (crypto.PrivKey, error) {
-	var r io.Reader
-	if seed == 0 {
-		r = crand.Reader
-	} else {
-		r = mrand.New(mrand.NewSource(seed))
-	}
```

Comment:
The new linter complained about the unsafe use of random. We didn't have any tests or other code that needed deterministic keys, so I just simplified it. The documentation at the top must have been written for a different iteration that took a directory instead of a full path.